modules/ingester/instance.go

```diff
@@ -332,7 +332,17 @@ func (i *instance) tracesToCut(cutoff time.Duration, immediate bool) []*trace {
 func (i *instance) writeTraceToHeadBlock(id common.ID, b []byte) error {
 	i.blocksMtx.Lock()
 	defer i.blocksMtx.Unlock()
-
+	if i.headBlock == nil {
```

Comment:
Any particular reason you added this?  I would prefer keeping the PR focused on the change to remove BucketExists.

Reply:
Yes. I will meet some CI errors if those code not add.

Reply:
Ah, I think @mdisibio discovered and is working on this issue.  Thank you for also trying to address it.